src/components/BButton/BButton.test.js

```diff
@@ -0,0 +1,63 @@
+import { mount, shallow } from 'vue-test-utils'
+import BButton from './BButton'
+
+describe('BButton', () => {
+  it('renders a button', () => {
+    const wrapper = mount(BButton, {
+      slots: {
+        default: 'hello world'
+      }
+    })
+
+    expect(wrapper.html()).toMatchSnapshot()
+  })
+
+  it('puts height and width props into style', () => {
+    const wrapper = shallow(BButton, {
+      propsData: {
+        height: '50px',
+        width: '100px'
+      }
+    })
+
+    expect(wrapper.hasStyle('height', '50px')).toBe(true)
+    expect(wrapper.hasStyle('width', '100px')).toBe(true)
+  })
+
+  it('is enabled when disabled prop is not passed', () => {
+    const wrapper = shallow(BButton, {})
+
+    expect(wrapper.props().disabled).toBe(false)
```

Comment:
Isn't this just testing that disabled didn't get passed in? It doesn't seem to be testing that the button is actually using the disabled value

Reply:
yeah I thought about that as I was writing it. Need to think more about how to test if it's actually disabled and vice versa. Maybe I can use the click function somehow.

Reply:
Testing if a button is disabled if it has `disabled` seems like too much. That sounds like a) we don't know how to write a Vue component or b) we are now testing actual HTML. If you correctly bind `disabled` the element should have it and should behave according to the HTML specs. In my opinion, a snapshot test works fine.